helix-lsp/src/lib.rs

```diff
@@ -129,7 +129,11 @@ pub mod util {
     ) -> Option<usize> {
         let pos_line = pos.line as usize;
         if pos_line > doc.len_lines() - 1 {
-            return None;
+            // If it extends past the end, truncate it to the end. This is because the
+            // way the LSP describes the range including the last newline is by
+            // specifying a line number after what we would call the last line.
+            log::warn!("LSP position {pos:?} out of range assuming EOF");
+            return Some(doc.len_chars());
```

Comment:
This is https://github.com/helix-editor/helix/issues/6022, right? And this change brings us in line with VSCode's handling of this edge-case?

Reply:
exactly